### DISCUSSION TITLE
Community leaders (project stewards) and updating our Code of Conduct

### DISCUSSION THREAD

**Discussion by martinvonz** [jj-vcs/jj#2227] [General]
Hi,

Our [current code of conduct](https://martinvonz.github.io/jj/v0.9.0/code-of-conduct.html) was part of the boilerplate I got from Google's Open-Source Programs Office. As you can see, I still haven't filled in names and emails for project stewards, since we simply don't have any project stewards yet. Since the community is getting larger, we should fix that.

Also, our code of conduct is based on [Contributor Covenant Code of Conduct 1.4](https://www.contributor-covenant.org/version/1/4/code_of_conduct/). There's now a [version 2.1](https://www.contributor-covenant.org/version/2/1/code_of_conduct/) of it. Assuming that there were good reasons for the changes made between those two versions, I'm leaning towards updating to the new version. Does that sound good? Any objections?

Does anyone volunteer to be a community leader (seems to be what's called a project steward in the 1.4 version)? I suspect we should ideally have at least two community leaders, so community members who have any issue with one of the community leaders can reach out to the other community leader. We also have the fallback of contacting [opensource@google.com](mailto:opensource@google.com), so maybe one is enough? I'm happy to hear thoughts from people who have more experience with this kind of stuff.

Thanks

**Comment by jaraco** [comment]
I'm willing to serve, but also happy to defer to anyone else interested. Presumably you'll be one of the leaders, Martin?

Who else is interested in being a project steward?

**Reply by martinvonz** [reply]
I can do it if no one else wants to (and it currently does seem like no one else does).

**Reply by avamsi** [reply]
I would volunteer but I tend to get to jj discussions (GitHub / Discord) in batches at times and I suspect this won't be ideal.

**Reply by martinvonz** [reply]
I think that's probably fine. I think there's some value in not having two people working so closely (Jason is my manager) as only community leaders. For example, if someone has a problem with me, they might feel more comfortable contacting you than contacting Jason.

**Reply by avamsi** [reply]
Sounds good, feel free to consider me in that case (like Jason, I'm also happy to defer to someone else more active / interested).

**Reply by martinvonz** [reply]
@jaraco and @avamsi, it sounds like it would be you two who would be community leaders then. I just want to confirm that you're both okay with that. I'm not sure what the next step is other than adding your email addresses where the placeholder currently is.

**Reply by jaraco** [reply]
SGTM. I'll plan to draft a PR but feel free to beat me to it.

**Reply by arxanas** [reply]
If we need another person, I can volunteer; like @avamsi, I tend to do my GitHub (and email) management in batches, but maybe two people is still better than one?

**Reply by martinvonz** [reply]
Thanks! Yes, it just seems better to have more people (except for the usual "shared responsibility is no responsibility" problem, but I'm not concerned about that yet). I'll let @jaraco draft the PR.

**Comment by jaraco** [comment]
I'm working on this today, and I'm thinking it would be best to have a mailing list (google group) for the updated CoC. I've created `jj-abuse@googlegroups.com`. I considered also creating a channel in Discord through which people could report concerns, but I'm unaware of a way to do that where only the community leaders and the reporter are privileged to the conversation.

**Reply by martinvonz** [reply]
Thanks!

>  I considered also creating a channel in Discord through which people could report concerns, but I'm unaware of a way to do that where only the community leaders and the reporter are privileged to the conversation.

Yes, I would think that it's not possible to create a channel that the admins (currently @spectral54 and I) could not spy on. So just a mailing list sounds good.

**Comment by jaraco** [comment]
PR at #2387.
